openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py

```diff
@@ -0,0 +1,67 @@
+"""PathstreamExternalEnrollment class file."""
+import logging
+from datetime import datetime
+
+from django.db import IntegrityError
+
+from openedx_external_enrollments.external_enrollments.base_external_enrollment import BaseExternalEnrollment
+from openedx_external_enrollments.models import ExternalEnrollment
+
+LOG = logging.getLogger(__name__)
+
+
+class PathstreamExternalEnrollment(BaseExternalEnrollment):
+    """
+    PathstreamExternalEnrollment class.
+    """
+    def __str__(self):
+        return 'pathstream'
+
+    def _get_enrollment_data(self, data):  # pylint: disable=arguments-differ
+        """
+        Returns a string with the data required to be treated as a log.
+        String format: 'course_key,email,date_time,status'
+        """
+        return '{course_key},{email},{date_time},{status}\n'.format(
+            course_key=data.get('course_id'),
+            email=data.get('user_email'),
+            date_time=datetime.utcnow(),
+            status=str(data.get('is_active')).lower(),
+        )
+
+    def _post_enrollment(self, data, course_settings=None):
+        """
+        Save enrollment data in ExternalEnrollment model.
+        """
+        LOG.info('calling enrollment for [%s] with data: %s', self.__str__(), data)
+        LOG.info('calling enrollment for [%s] with course settings: %s', self.__str__(), course_settings)
+
+        try:
+            enrollment, created = ExternalEnrollment.objects.get_or_create(  # pylint: disable=no-member
+                controller_name=str(self),
+                course_shell_id=data.get('course_id'),
+                email=data.get('user_email'),
+            )
```

Comment:
Since you are saving the model anyways, you can use update_or_create, does it make sense @Jacatove?
```suggestion
            enrollment, created = ExternalEnrollment.objects.update_or_create(  # pylint: disable=no-member
                controller_name=str(self),
                course_shell_id=data.get('course_id'),
                email=data.get('user_email'),
                defaults={
                    'meta': {
                        'enrollment_data_formated': self._get_enrollment_data(data),
                        'is_uploaded': False,
                    },
                },
            )
```

Reply:
@ivanvgh If it updates the meta, we will replace the old enrollment data by the new one. Appending the new data to meta is required to persist all events.

Reply:
Do we have a use case where the enrollment happens twice or needs to save the event as you say?

Reply:
@ivanvgh After an enrollment event, an unenrollment event will trigger another external enrollment for same user and course.